### ISSUE TITLE
Error with dependency package `twofish`

### ISSUE THREAD

**Issue by wqhanginge** [blluv/pgmm_decrypt#5]
I encountered a `ModuleNotFoundError` when using this repository with **Python 3.12**. Here is the error message I received:

```sh
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\wang\AppData\Roaming\Python\Python312\site-packages\pgmm_decrypt\__init__.py", line 1, in <module>
    from .pgmm import decrypt_pgmm_key, decrypt_pgmm_resource
  File "C:\Users\wang\AppData\Roaming\Python\Python312\site-packages\pgmm_decrypt\pgmm.py", line 1, in <module>
    from twofish import Twofish
  File "C:\Users\wang\AppData\Roaming\Python\Python312\site-packages\twofish.py", line 11, in <module>
    import imp
ModuleNotFoundError: No module named 'imp'
```

I read the [documentation](https://docs.python.org/3.11/library/imp.html#module-imp) for the `imp` module on the Python website, and it says that `imp` is **deprecated** and will be removed in **Python 3.12**. It seems that `twofish` uses a deprecated module which no longer exists in Python 3.12 and later. What’s worse, the GitHub repository (https://github.com/keybase/python-twofish) of `twofish` has had no new commits for 11 years. Therefore, this external issue is unlikely to be fixed in the foreseeable future.

My question is, **do you plan to fix this issue?** It wouldn’t be too much work to fork the `twofish` repo here and fix it, or a better choice might be to build a new C extension from the source code of `twofish` for this repo.

I could help whether you want to fork the repo or build a C extension. :)

**Comment by blluv** [comment]
Okay. I've moved twofish to pgmm_decrypt. However, I don't have the pgmm game resources, so I can't test it. If you could send the game files to biluv@protonmail.com, I'll test it and then merge (and I plan to implement automatic weak detection).

https://github.com/blluv/pgmm_decrypt/tree/twofish

**Comment by wqhanginge** [comment]
Thanks. I've sent you the encrypted game files I have so far. I hope they will be helpful.

**Comment by blluv** [comment]
This issue has been resolved in #6
